### PR TITLE
Don't allow tokens in paths.

### LINKED ISSUE #1764: Token is exposed as plaintext in the audit backend when is passed in the URL
In case of accessing the auth/token/lookup api
the token is logged as plaintext (as it's in the url) to the audit backend

example of request:

Aug 23 09:31:45 ip-10-0-3-185 vault[1]: {"time":"2016-08-23T09:31:45Z","type":"request","auth":{"display_name":"root","policies":["root"],"metadata":null},"request":{"operation":"read","client_token":"hmac-sha256:1024397ba71fcde0b0d933c0b79c667d217043b14ffaea86f47b94cd6326bc59","path":"auth/token/lookup/accaadc2-7b6e-c929-d594-2d8de6d18c47","data":null,"remote_address":"10.0.6.92","wrap_ttl":0},"error":""}

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #1764

### CODE REVIEW

File: api/auth_token.go [hunk 1]
```diff
@@ -77,10 +86,11 @@ func (c *TokenAuth) LookupSelf() (*Secret, error) {
 }
 
 func (c *TokenAuth) Renew(token string, increment int) (*Secret, error) {
-	r := c.c.NewRequest("PUT", "/v1/auth/token/renew/"+token)
-
-	body := map[string]interface{}{"increment": increment}
-	if err := r.SetJSONBody(body); err != nil {
+	r := c.c.NewRequest("PUT", "/v1/auth/token/renew")
```
Comment (vishalnayak) [inline api/auth_token.go]: Can we make this `POST` as well, just to be consistent?

File: api/auth_token.go [hunk 2]
```diff
@@ -126,7 +141,13 @@ func (c *TokenAuth) RevokeAccessor(accessor string) error {
 // RevokeOrphan revokes a token without revoking the tree underneath it (so
 // child tokens are orphaned rather than revoked)
 func (c *TokenAuth) RevokeOrphan(token string) error {
-	r := c.c.NewRequest("PUT", "/v1/auth/token/revoke-orphan/"+token)
+	r := c.c.NewRequest("PUT", "/v1/auth/token/revoke-orphan")
```
Comment (vishalnayak) [inline api/auth_token.go]: Same here

File: api/auth_token.go [hunk 3]
```diff
@@ -152,7 +173,13 @@ func (c *TokenAuth) RevokeSelf(token string) error {
 // the entire tree underneath -- all of its child tokens, their child tokens,
 // etc.
 func (c *TokenAuth) RevokeTree(token string) error {
-	r := c.c.NewRequest("PUT", "/v1/auth/token/revoke/"+token)
+	r := c.c.NewRequest("PUT", "/v1/auth/token/revoke")
```
Comment (vishalnayak) [inline api/auth_token.go]: Same here

File: api/auth_token.go (hunk 1 above)
Comment (jefferai) [inline api/auth_token.go]: `PUT` is actually more widely used. However, I actually like having both in the API, as it serves as an early smoke test if we accidentally break PUT/POST compatibility.

File: vault/token_store.go [hunk 4]
```diff
@@ -295,24 +286,21 @@ func NewTokenStore(c *Core, config *logical.BackendConfig) (*TokenStore, error)
 				},
 
 				Callbacks: map[logical.Operation]framework.OperationFunc{
-					logical.ReadOperation: t.handleLookupSelf,
+					logical.ReadOperation:   t.handleLookupSelf,
```
Comment (vishalnayak) [inline vault/token_store.go]: Can we remove the `ReadOperation` from here?

File: api/auth_token.go (hunk 1 above)
Comment (vishalnayak) [inline api/auth_token.go]: Sounds good.

**Comment by vishalnayak** [comment]
LGTM!

File: vault/token_store.go (hunk 4 above)
Comment (jefferai) [inline vault/token_store.go]: We can but I left it in for backwards compatibility, as it's not a security risk.

**Pull request merged**